src/event.cr

```diff
@@ -80,26 +80,74 @@ module Event
       LibEvent2.event_base_loopbreak(@base)
     end
 
-    def new_dns_base(init = true)
-      DnsBase.new LibEvent2.evdns_base_new(@base, init ? 1 : 0)
+    def to_unsafe
+      @base
     end
   end
 
-  struct DnsBase
-    def initialize(@dns_base : LibEvent2::DnsBase)
+  # :nodoc:
+  class DnsBase
+    # :nodoc:
+    # TODO: consider using a struct & pass pointer
+    class Response
+      property result : Int32?
+      property addrinfo : Pointer(LibEvent2::EvutilAddrinfo)?
+
+      def initialize(@fiber : Fiber)
+      end
+
+      def resume
+        @fiber.resume
+      end
+
+      def cancelled?
+        @result == LibEvent2::EVUTIL_EAI_CANCEL
+      end
     end
 
-    def getaddrinfo(nodename, servname, hints, data, &callback : LibEvent2::DnsGetAddrinfoCallback)
-      request = LibEvent2.evdns_getaddrinfo(@dns_base, nodename, servname, hints, callback, data.as(Void*))
-      GetAddrInfoRequest.new request if request
+    def initialize(base : Base)
+      @dns_base = LibEvent2.evdns_base_new(base, 1)
     end
 
-    struct GetAddrInfoRequest
-      def initialize(@request : LibEvent2::DnsGetAddrinfoRequest)
+    def finalize
+      LibEvent2.evdns_base_free(@dns_base, 1)
+    end
+
+    def getaddrinfo(domain, service, hints, timeout = nil, &block)
```

Comment:
I would expect this because I have used getaddrinfo but is there any merit to using snakecase `get_address_info`?